woocommerce/woocommerce
# API tests for CRUD methods in Orders API

### All Submissions:


### Changes proposed in this Pull Request:

Closes the following issues:
- #30670
- #30671 
- #30672 

The tests in this PR only includes the following API's:
- Create an order
- Update an order
- Retrieve an order
- Delete an order

I did not include tests for the "Batch update orders" endpoint yet, because the test file seems to be getting lengthy already.

### How to test the changes in this Pull Request:

1. Clone this repo and run the setup scripts for the e2e environment.
2. Run the orders API tests: `npm test -- --group=orders`
3. You should see the new `Orders API tests: CRUD` tests pass:
```
 PASS  tests/orders/orders-crud.test.js
  Orders API tests: CRUD
    Create an order
      ✓ can create a pending order by default (225ms)
      ✓ can create an order with status 'pending' (208ms)
      ✓ can create an order with status 'processing' (246ms)
      ✓ can create an order with status 'on-hold' (235ms)
      ✓ can create an order with status 'completed' (234ms)
      ✓ can create an order with status 'cancelled' (180ms)
      ✓ can create an order with status 'refunded' (181ms)
      ✓ can create an order with status 'failed' (188ms)
    Retrieve an order
      ✓ can retrieve an order (94ms)
    Update an order
      ✓ can update status of an order to 'pending' (109ms)
      ✓ can update status of an order to 'processing' (155ms)
      ✓ can update status of an order to 'on-hold' (116ms)
      ✓ can update status of an order to 'completed' (139ms)
      ✓ can update status of an order to 'cancelled' (116ms)
      ✓ can update status of an order to 'refunded' (125ms)
      ✓ can update status of an order to 'failed' (119ms)
      ✓ can add shipping and billing contacts to an order (152ms)
      ✓ can add a product to an order (138ms)
      ✓ can pay for an order (263ms)
    Delete an order
      ✓ can permanently delete an order (242ms)
```
4. Verify either through the UI, or through the "List all products" and "List all orders" API that no products and orders were left over by the tests.

### Other information:


### Changelog entry

> Added: CRUD tests for the Orders API

### FOR PR REVIEWER ONLY:
